multiwallet.go

```diff
@@ -413,6 +413,14 @@ func (mw *MultiWallet) saveNewWallet(wallet *Wallet, setupWallet func() error) (
 		}
 
 		walletDataDir := filepath.Join(mw.rootDir, strconv.Itoa(wallet.ID))
+
+		dirExists, err := fileExists(walletDataDir)
+		if err != nil {
+			return err
+		} else if dirExists {
+			os.RemoveAll(walletDataDir)
+		}
```

Comment:
This is a little disturbing, I'd wager that if a directory exists at this path, then it would hold a wallet database file. Deleting wallet files without user permission / awareness is concerning.

Also, the dir name used for new wallets is a number that's properly incremented by the db to prevent collision, it doesn't even reuse a number even if the wallet that was there is deleted, so we typically should not have this issue, but if we do, I think aborting the operation is the preferable way to go, the user should manually delete the directory that is posing trouble.

It's even possible that the next try from the user won't have that issue because the wallet id would be again incremented.

Reply:
If a panic occurs when creating a wallet due to memory, the wallet folder will remain and the wallet loader won't be able to create the wallet. For mobile, the folders are fully managed by the apps so there should be no scenario of an undocumented folder making it there unless it happened due to panic.

Reply:
Given that `dcrlibwallet` could be used outside mobile devices, or even in mobile devices with a publicly accessible `rootDir`, the assumption that orphaned wallet dirs are a result of a previous panic becomes a rough assumption at best.

Since `dcrlibwallet` does not ensure that the `rootDir` cannot be modified externally, we should assume that a user could store some valuable information in the conflicting directory and return an error instead of deleting that directory.